src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -73,13 +76,13 @@ private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
                     var oldDocument = e.OldSolution.GetDocument(e.DocumentId);
                     var newDocument = e.NewSolution.GetDocument(e.DocumentId);
 
-                    var oldTextVersion = oldDocument.GetTextVersionAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None);
-                    var newTextVersion = newDocument.GetTextVersionAsync(CancellationToken.None).WaitAndGetResult(CancellationToken.None);
-
-                    if (oldTextVersion != newTextVersion)
+                    if (documentChangedWorker == null)
```

Comment:
Usual pattern I think is just to have a placeholder task that's always complete. We have SpecializedTasks.EmptyTask to help with this.